### PR TITLE
the mother of all pull requests

### PR DESCRIPTION
**Pull request by despair86**
:DDDDD

all of the platform-specific code is now in the platform headers

also includes packaging fluff

### CODE REVIEW

**Comment by neuroscr** [comment]
and where was DNSd doing that? I don't see any changes or fixes to DNSd here.

**Comment by despair86** [comment]
i just found out about it last night, before this, the stack would end up completely blasted to bits

File: test/test_dns_unit.cpp [hunk 1]
```diff
@@ -68,7 +68,7 @@ TEST_F(DNSTest, TestDecodeDNSstring)
 TEST_F(DNSTest, TestCodeDomain)
 {
   char buffer[16];
-  llarp::Zero(&buffer, 15);
+  llarp::Zero(&buffer, 16);
```
Comment (majestrate) [inline test/test_dns_unit.cpp]: should probably be `buffer` not `&buffer`

Comment (despair86) [reply test/test_dns_unit.cpp]: fixed

**Comment by majestrate** [comment]
this ready now?

**Pull request merged**